android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java

```diff
@@ -375,17 +380,26 @@ private void downloadDeviceIcon(final String url){
 		new Thread(new Runnable(){
 			@Override
 			public void run(){
-				try{
-					deviceLogo = AndroidTools.downloadImage(url);
+				mLockScreenDeviceIconManager.retrieveIcon(url, new LockScreenDeviceIconManager.OnIconRetrievedListener() {
+					@Override
+					public void onImageRetrieved(Bitmap icon) {
```

Comment:
I don't think that will work. We are retrieving the icon in an async way and immediately use the icon without waiting for onImageRetrieved () to be called. I think deviceLogo will always be null. I suggest moving the code after the async call to be inside the `onImageRetrieved()` callback.

Reply:
moved